### PR TITLE
Fix issue where page crashes due to CORs error

### PR DESCRIPTION
**Pull request by ashera96**
### Purpose

- Resolves https://github.com/wso2/api-manager/issues/2835

Fixes the issue where page crash when encountered with a CORS issue

### CODE REVIEW

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx [hunk 1]
```diff
@@ -176,7 +177,7 @@ const ApiChat = () => {
                 setSpecEnrichmentError(
                     intl.formatMessage({
                         id: 'Apis.Details.ApiChat.components.specEnrichmentError.genericError',
-                        defaultMessage: 'An error occurred when loading API Chat.',
+                        defaultMessage: 'Error occurred when loading API Chat.',
```
Comment (lasanthaS) [inline portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx:180]: Error occurred "**while**" loading API Chat.

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx [hunk 2]
```diff
@@ -417,7 +418,7 @@ const ApiChat = () => {
 
     const invokeAPI = async (generatedRequest) => {
         const { method, path, inputs } = generatedRequest;
-        const { parameters, requestBody } = inputs || {};
+        const { parameters = {}, requestBody = {} } = inputs || {};
```
Comment (lasanthaS) [inline portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx:421]: space between curly braces `{ }`

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx [hunk 3]
```diff
@@ -158,21 +158,54 @@ const ApiChatResponse: React.FC<ApiChatResponseProps> = ({
...
+     *
+     * @param {string} str Response body.
+     * @returns {string} Content type of the response.
+     */
+    const inferContentType = (str: string) => {
+        const trimmedStr = str.trim();
+        const xmlRegex = /^\s*<[^>]+>/;
+        const jsonRegex = /^[\\{\\[](.*?)[\\}\]]$/;
+
+        if (xmlRegex.test(trimmedStr)) {
```
Comment (lasanthaS) [inline portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx:172]: Can simplify as follows.

```javascript
if (xmlRegex.test(trimmedStr)) {
    return 'application/xml';
}

if (jsonRegex.test(trimmedStr)) {
    try {
        JSON.parse(trimmedStr);
        return 'application/json';
    } catch (error) {
        // Handle potential invalid JSON structure
    }
} 
return 'text/plain';

```

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx [hunk 4]
```diff
@@ -158,21 +158,54 @@ const ApiChatResponse: React.FC<ApiChatResponseProps> = ({
...
     /**
      * Renders the execution result body.
      *
      * @param {any} executionResult Execution result to render.
      * @returns {JSX.Element} Execution result body to render.
      */
     const renderExecutionResultBody = (executionResult: any) => {
-        const contentType = executionResult.headers.get('Content-Type');
+        // Determine content type
+        let contentType = 'application/json';
```
Comment (lasanthaS) [inline portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx:194]: Let's move these content types as constants

**Review by lasanthaS — CHANGES_REQUESTED** [review]

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx (hunk 2 above)
Comment (lasanthaS) [reply portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx:421]: Ignore this please as this is fine with the ESLint rules

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx (hunk 4 above)
Comment (ashera96) [reply portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx]: Fixed via https://github.com/wso2/apim-apps/pull/689/commits/70f944fc914b0bc980fede3152bf3dedd7450ae2

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx (hunk 3 above)
Comment (ashera96) [reply portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/ApiChatResponse.tsx:176]: Fixed via https://github.com/wso2/apim-apps/pull/689/commits/70f944fc914b0bc980fede3152bf3dedd7450ae2

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx (hunk 1 above)
Comment (ashera96) [reply portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx]: Fixed via https://github.com/wso2/apim-apps/pull/689/commits/70f944fc914b0bc980fede3152bf3dedd7450ae2

**Review by lasanthaS — APPROVED** [review]

**Pull request merged**